lib/rets/metadata/file_cache.rb

```diff
@@ -0,0 +1,29 @@
+module Rets
+  module Metadata
+
+    # This metadata cache persists the metadata to a file.
+    class FileCache
+
+      def initialize(path)
+        @path = path
+      end
+
+      # Save the metadata.
+      def save(metadata)
+        File.open(@path, "wb") do |file|
+          Marshal.dump(metadata, file)
+        end
+      end
+
+      # Load the metadata.  Return nil if it could not be loaded.
+      def load
+        File.open(@path, "rb") do |file|
+          Marshal.load(file)
+        end
+      rescue IOError, SystemCallError
+        nil
+      end
```

Comment:
Rather than persisting the `Metadata::Root` object I think it might be nicer to persist the `sources`.

In our codebase we have similar code that does:

```
File.open(@path, "wb") do |file|
  Marshal.dump(metadata.marshal_dump, file)
end
```

```
File.open(@path, "rb") do |file|
  Metadata::Root.new(Marshal.load(file), logger)
end
```

That way we're not persisting the logger at all.

The downside is the tree structure has to be rebuilt, whilst I've not done any perf checks we haven't found it to be a problem.